src/ape/api/providers.py

```diff
@@ -1032,7 +1032,19 @@ def get_receipt(
         except TimeExhausted as err:
             raise TransactionNotFoundError(txn_hash) from err
 
-        txn = dict(self.web3.eth.get_transaction(HexStr(txn_hash)))
+        network_config = self.network.config.get(self.network.name)
+        max_retries = network_config.get("max_receipt_retries")
```

Comment:
`get()` implies this could return `None`, which it would do if the network config was missing the property.
so we need a default value.

let's set a constant (same place other constants are set like `DEFAULT_LOCAL_TRANSACTION_ACCEPTANCE_TIMEOUT`) and use that value both here as the default:

```
network_config = self.network.config.get(self.network.name, DEFAULT_MAX_RETRIES_TX)
```

as well as in the network config's default:

```python
max_receipt_retries: int = DEFAULT_MAX_RETRIES_TX
```